### PR TITLE
Having fix in inventory and Discovery workflow manager modules

### PR DESCRIPTION
**Pull request by madhansansel**
Addressing the following issues in the below workflow manager modules

inventory_workflow_manger module:
a) Edit device with role stuck in infinite loop
b) Untagging a golden image in a child site should return failed or print a message.
c) Adding a device with 'snmp_version: v2' failed
d) Result messages missing or incorrect when deleting multiple devices
e) Edit device with snmp_auth_passphrase, snmp_priv_passphrase not changed
f) Inconsistent 'enable_password' requirement for adding devices to inventory

discovery_workflow_manager Module:
a) Module first time run fails even though discovery started successfully
b) Any of SNMP credentials type snmpv2 or v3) should work for discovery.
c) Discover devices when SNMPv2 credentials are not present
d) Discovery failed intermittently, relating to different states of the discovery job
e) Error handling with proper meesage when discovery name is not present

### CODE REVIEW

**Comment by madhansansel** [comment]
Hello @fmunozmiranda and @wastorga , Could you please review and approve the changes?

File: plugins/modules/discovery_intent.py [hunk 1]
```diff
@@ -204,13 +211,6 @@
                     - Requires valid SSH credentials to work.
                     - Avoid standard ports like 22, 80, and 8080.
                 type: str
-      use_global_cred:
```
Comment (wastorga) [inline plugins/modules/discovery_intent.py]: use_global_cred has been renamed to use_global_credentials. This too should be documented, at least in the `notes` section of the module.

File: plugins/modules/discovery_workflow_manager.py [hunk 2]
```diff
@@ -359,7 +359,7 @@
 
 EXAMPLES = r"""
 - name: Execute discovery devices with both global credentials and discovery specific credentials
-  cisco.dnac.discovery_workflow_manager:
+  cisco.dnac.discovery_intent:
```
Comment (wastorga) [inline plugins/modules/discovery_workflow_manager.py:362]: Is this the right name?

File: plugins/modules/discovery_intent.py [hunk 3]
```diff
@@ -200,34 +204,122 @@
...
         type: str
-        required: true
+        default: ssh
       retry:
         description: Number of times to try establishing connection to device
         type: int
       timeout:
         description: Time to wait for device response in seconds
         type: int
-      global_cli_len:
```
[the remark this answers is not in the document]
Comment (madhansansel) [reply plugins/modules/discovery_intent.py:227]: Done

File: changelogs/changelog.yaml [hunk 4]
```diff
@@ -806,3 +806,10 @@ releases:
           - The 'discovery_workflow_manager' module streamlines the discovery of devices using various methods including single IP, IP range, multi-range, CDP, CIDR, and LLDP. It also offers the ability to clear out discoveries by deleting them from the discovery database, with an option to delete all discoveries simultaneously.
           - The 'provision_workflow_manager' module provisions and re-provisions devices added in the inventory to site, by taking management IP address as input. It allows provisioning of both wired and wireless devices. It also allows un-provisioning of devices.
           - The 'template_workflow_manager' module is responsible for overseeing templates, export projects/templates, and import projects/templates. It handles configuration templates by enabling the creation, updating, and deletion of templates and projects. Additionally, the module supports export functionality to retrieve project and template details from Cisco Catalyst Center, and Import functionality to create templates and projects within the Cisco Catalyst Center.
+    6.11.1:
```
[the remark this answers is not in the document]
Comment (madhansansel) [reply changelogs/changelog.yaml]: Done.

File: plugins/modules/discovery_workflow_manager.py [hunk 5]
```diff
@@ -200,34 +204,122 @@
...
         type: str
-        required: true
+        default: ssh
       retry:
         description: Number of times to try establishing connection to device
         type: int
       timeout:
         description: Time to wait for device response in seconds
         type: int
-      global_cli_len:
```
[the remark this answers is not in the document]
Comment (madhansansel) [reply plugins/modules/discovery_workflow_manager.py:227]: Taken care.

File: plugins/modules/inventory_workflow_manager.py [hunk 6]
```diff
@@ -262,34 +259,97 @@
...
-            type: int
-          interface_gateway:
-            description: The name identifier for the gateway associated with the interface.
+          site_name:
+            description: Indicates the exact location where the wired device will be provisioned. This is a string value that should
+                represent the complete hierarchical path of the site (For example, "Global/USA/San Francisco/BGL_18/floor_pnp").
             type: str
-          lag_or_port_number:
-            description: The Link Aggregation Group (LAG) number or port number assigned to the interface.
+          resync_retry_count:
```
[the remark this answers is not in the document]
Comment (madhansansel) [reply plugins/modules/inventory_workflow_manager.py:278]: Taken care

File: plugins/modules/inventory_workflow_manager.py [hunk 7]
```diff
@@ -262,34 +259,97 @@
...
-            description: The VLAN (Virtual Local Area Network) ID associated with the network interface.
+            default: 200
+          resync_retry_interval:
+            description: Sets the interval, in seconds, at which the system will recheck the device status throughout the provisioning
+                process. If unspecified, the system will check the device status every 2 seconds by default.
             type: int
-          interface_name:
-            description: Name of the interface.
+            default: 2
+      reprovision_wired_device:
```
[the remark this answers is not in the document]
Comment (madhansansel) [reply plugins/modules/inventory_workflow_manager.py:290]: Taken care.

File: plugins/modules/inventory_workflow_manager.py [hunk 8]
```diff
@@ -262,34 +259,97 @@
...
+        suboptions:
+          device_ip:
+            description: Specifies the IP address of the wired device. This is a string value that should be in the format of
+                standard IPv4 or IPv6 addresses.
+            type: str
+          site_name:
+            description: Indicates the exact location where the wired device will be provisioned. This is a string value that should
+                represent the complete hierarchical path of the site (For example, "Global/USA/San Francisco/BGL_18/floor_pnp").
             type: str
+      provision_wireless_device:
```
[the remark this answers is not in the document]
Comment (madhansansel) [reply plugins/modules/inventory_workflow_manager.py:305]: Taken care..

File: plugins/modules/discovery_intent.py (hunk 1 above)
Comment (madhansansel) [reply plugins/modules/discovery_intent.py]: Taken care.

**Review by wastorga — APPROVED** [review]
The attribute itself could **also** have a `version_added: 6.12.0` added in its documentation.
And for renames and deletions `notes:` could be used to have those changes inside the plugin/modules documentation itself.

Nonetheless, I think adding it here in changelogs/changelog.yaml is enough to meet Ansible review requirements.

**Pull request merged**